### PR TITLE
Require explicit enabling of RGB Matrix modes

### PR DESCRIPTION
**Pull request by drashna**
## Description

As title

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -482,48 +482,48 @@ You can disable a single effect by defining `DISABLE_[EFFECT_NAME]` in your `con
...
+|`#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL`            |Enables `RGB_MATRIX_CYCLE_PINWHEEL`           |
+|`#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL`              |Enables `RGB_MATRIX_CYCLE_SPIRAL`             |
+|`#define ENABLE_RGB_MATRIX_DUAL_BEACON`               |Enables `RGB_MATRIX_DUAL_BEACON`              |
+|`#define ENABLE_RGB_MATRIX_RAINBOW_BEACON`            |Enables `RGB_MATRIX_RAINBOW_BEACON`           |
+|`#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS`         |Enables `RGB_MATRIX_RAINBOW_PINWHEELS`        |
+|`#define ENABLE_RGB_MATRIX_RAINDROPS`                 |Enables `RGB_MATRIX_RAINDROPS`                |
+|`#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS`       |Enables `RGB_MATRIX_JELLYBEAN_RAINDROPS`      |
+|`#define ENABLE_RGB_MATRIX_HUE_BREATHING`             |Enables `RGB_MATRIX_HUE_BREATHING`            |
+|`#define ENABLE_RGB_MATRIX_HUE_PENDULUM`              |Enables `RGB_MATRIX_HUE_PENDULUM`             |
+|`#define ENABLE_RGB_MATRIX_HUE_WAVE `                 |Enables `RGB_MATRIX_HUE_WAVE `                |
```
Comment (Xelus22) [inline docs/feature_rgb_matrix.md:510]: extra space here :D

File: docs/feature_rgb_matrix.md [hunk 2]
```diff
@@ -482,48 +482,48 @@ You can disable a single effect by defining `DISABLE_[EFFECT_NAME]` in your `con
...
+|`#define ENABLE_RGB_MATRIX_FRACTAL`                   |Enables `RGB_MATRIX_FRACTAL`                  |
+|`#define ENABLE_RGB_MATRIX_PIXEL_RAIN `               |Enables `RGB_MATRIX_PIXEL_RAIN `              |
+|`#define ENABLE_RGB_MATRIX_TYPING_HEATMAP`            |Enables `RGB_MATRIX_TYPING_HEATMAP`           |
+|`#define ENABLE_RGB_MATRIX_DIGITAL_RAIN`              |Enables `RGB_MATRIX_DIGITAL_RAIN`             |
+|`#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE`     |Enables `RGB_MATRIX_SOLID_REACTIVE_SIMPLE`    |
```
Comment (Xelus22) [inline docs/feature_rgb_matrix.md:515]: maybe put in the docs somewhere `#define RGB_MATRIX_KEYPRESSES` must be enabled to compile these effects. Same for `#define RGB_MATRIX_FRAMEBUFFER_EFFECTS`

File: docs/feature_rgb_matrix.md [hunk 3]
```diff
@@ -482,48 +482,48 @@ You can disable a single effect by defining `DISABLE_[EFFECT_NAME]` in your `con
...
+|`#define ENABLE_RGB_MATRIX_DUAL_BEACON`               |Enables `RGB_MATRIX_DUAL_BEACON`              |
+|`#define ENABLE_RGB_MATRIX_RAINBOW_BEACON`            |Enables `RGB_MATRIX_RAINBOW_BEACON`           |
+|`#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS`         |Enables `RGB_MATRIX_RAINBOW_PINWHEELS`        |
+|`#define ENABLE_RGB_MATRIX_RAINDROPS`                 |Enables `RGB_MATRIX_RAINDROPS`                |
+|`#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS`       |Enables `RGB_MATRIX_JELLYBEAN_RAINDROPS`      |
+|`#define ENABLE_RGB_MATRIX_HUE_BREATHING`             |Enables `RGB_MATRIX_HUE_BREATHING`            |
+|`#define ENABLE_RGB_MATRIX_HUE_PENDULUM`              |Enables `RGB_MATRIX_HUE_PENDULUM`             |
+|`#define ENABLE_RGB_MATRIX_HUE_WAVE `                 |Enables `RGB_MATRIX_HUE_WAVE `                |
+|`#define ENABLE_RGB_MATRIX_FRACTAL`                   |Enables `RGB_MATRIX_FRACTAL`                  |
+|`#define ENABLE_RGB_MATRIX_PIXEL_RAIN `               |Enables `RGB_MATRIX_PIXEL_RAIN `              |
```
Comment (filterpaper) [inline docs/feature_rgb_matrix.md:512]:
```suggestion
|`#define ENABLE_RGB_MATRIX_PIXEL_RAIN`                |Enables `RGB_MATRIX_PIXEL_RAIN`              |
```
Sorry for missing this on my rebase.

**Review by filterpaper — CHANGES_REQUESTED** [review]

File: docs/feature_rgb_matrix.md (hunk 2 above)
Comment (drashna) [reply docs/feature_rgb_matrix.md:515]: I think I'll break up the table, then.  That way, it's much clearer

**Review by tzarc — APPROVED** [review]

**Comment by peepeetee** [comment]
I think there's a mistake that's made: in some boards ENABLE_RGB_MATRIX_PIXEL_FRACTAL was erroneously renamed to ENABLE_RGB_MATRIX_FRACTAL. See, for example, kprepublic/bm80hsrgb and 3keyecosystem/2key2

**Comment by filterpaper** [comment]
> I think there's a mistake that's made: in some boards ENABLE_RGB_MATRIX_PIXEL_FRACTAL was erroneously renamed to ENABLE_RGB_MATRIX_FRACTAL. See, for example, kprepublic/bm80hsrgb and 3keyecosystem/2key2

I should probably fix them. They were due to 'race condition' PR changes that were making their way into the last breaking change.

**Pull request merged**